bigquery/google/cloud/bigquery/_helpers.py

```diff
@@ -467,3 +467,20 @@ def to_api_repr(self):
         if self.name is not None:
             resource['name'] = self.name
         return resource
+
+
+class QueryParametersProperty(object):
+    """Custom property type, holding query parameter instances."""
+
+    def __get__(self, instance, owner):
+        """Descriptor protocol:  accessor"""
+        if instance is None:
+            return self
+        return list(instance._query_parameters)
+
+    def __set__(self, instance, value):
+        """Descriptor protocol:  mutator"""
+        if not all(isinstance(u, AbstractQueryParameter) for u in value):
```

Comment:
Can you replace `u` with something else? I don't even know what types `u` or `value` are supposed to be (other than `value` is iterable).